hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/BucketArgs.java

```diff
@@ -89,6 +91,36 @@ private BucketArgs(Boolean versioning, StorageType storageType,
     this.quotaInNamespace = quotaInNamespace;
   }
 
+  /**
+   * Private constructor, constructed via builder.
+   * @param versioning Bucket version flag.
+   * @param storageType Storage type to be used.
+   * @param acls list of ACLs.
+   * @param metadata map of bucket metadata
+   * @param bucketEncryptionKey bucket encryption key name
+   * @param sourceVolume
+   * @param sourceBucket
+   * @param quotaInBytes Bucket quota in bytes.
+   * @param quotaInNamespace Bucket quota in counts.
+   */
+  @SuppressWarnings("parameternumber")
+  private BucketArgs(Boolean versioning, StorageType storageType,
+      List<OzoneAcl> acls, Map<String, String> metadata,
+      String bucketEncryptionKey, String sourceVolume, String sourceBucket,
+      long quotaInBytes, long quotaInNamespace,
+      ECReplicationConfig ecReplicationConfig) {
+    this.acls = acls;
+    this.versioning = versioning;
+    this.storageType = storageType;
+    this.metadata = metadata;
+    this.bucketEncryptionKey = bucketEncryptionKey;
+    this.sourceVolume = sourceVolume;
+    this.sourceBucket = sourceBucket;
+    this.quotaInBytes = quotaInBytes;
+    this.quotaInNamespace = quotaInNamespace;
+    this.ecReplicationConfig = ecReplicationConfig;
+  }
+
```

Comment:
This constructor is private, so I think we can just add another parameter rather than duplicating?